src/cljs/athens/page.cljs

```diff
@@ -70,14 +70,37 @@
        [:div {:style {:margin-left 20}}
         [render-blocks (:block/uid @node)]]])))
 
+(defn title [title]
+  (let [s (reagent/atom {:editing false
+                         :current-title title
+                         :new-title title})
+        save! (fn [] (dispatch [:node/rename (:current-title @s) (:new-title @s)])
+                     (swap! s assoc :editing false))
+        cancel! (fn [] (swap! s #(-> %
+                                     (assoc :editing false)
+                                     (assoc :new-title (:current-title @s)))))]
+    (fn [title]
+      (if (:editing @s)
+        [:input {:value (:new-title @s)
+                 :auto-focus true
+                 :on-change #(swap! s assoc :new-title (-> % .-target .-value))
+                 :on-blur save!
+                 :on-key-down #(case (.-which %)
+                                 13 (save!) ; save on enter
+                                 27 (cancel!) ; cancel on esc
```

Comment:
Instead of adding the comments, might make more sense to define the codes somewhere:
```
(def enter-keycode 13)
(def esc-keycode 27)
```
 This can be reused by other parts of code and is more robust than the comment (imagine appending a 1 to the 13 in a big pull request and this is missed by the reviewers)